service/tbc/peermanager_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2024 Hemi Labs, Inc.
+// Use of this source code is governed by the MIT License,
+// which can be found in the LICENSE file.
+
+package tbc
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/wire"
+)
+
+func ping(ctx context.Context, t *testing.T, p *peer) error {
+	err := p.write(time.Second, wire.NewMsgPing(uint64(time.Now().Unix())))
+	if err != nil {
+		return err
+	}
+
+	for {
+		msg, _, err := p.read(time.Second)
+		if errors.Is(err, wire.ErrUnknownMessage) {
+			continue
+		} else if err != nil {
+			return err
+		}
+		switch msg.(type) {
+		case *wire.MsgPong:
+			return nil
+		}
+	}
+}
+
+func TestPeerManager(t *testing.T) {
+	t.Skip("this test connects to testnet3")
+	want := 2
+	wantLoop := want * 2
+	pm, err := NewPeerManager(wire.TestNet3, []string{}, want)
```

Comment:
@marcopeereboom does this connect to testnet3 during the test? it seemed like it did when I ran it so I skipped this as I don't want our tests to depend on testnet3 itself.

Reply:
yeah i think that's the bug. Need to provide 127.0.0.1 in seeds